File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/PersonForm.ts
import { StringEditor, LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface PersonForm {
    FirstName: StringEditor;
    LastName: StringEditor;
    GenderId: LookupEditor;
}

export class PersonForm extends PrefixedContext {
    static readonly formKey = 'CrmDB.Person';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PersonForm.init)  {
            PersonForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;

            initFormType(PersonForm, [
                'FirstName', w0,
                'LastName', w0,
                'GenderId', w1
            ]);
        }
    }
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/Person/PersonForm.cs
using Serenity.ComponentModel;

namespace CultureLinkCRM.CrmDB.Forms;

[FormScript("CrmDB.Person")]
[BasedOnRow(typeof(PersonRow), CheckNames = true)]
public class PersonForm
{
    public string FirstName { get; set; }
    public string LastName { get; set; }

    public int GenderId { get; set; }
}